object-manipulation-2/src/defaults.js

```diff
@@ -0,0 +1,15 @@
+/* exported defaults */
+
+function defaults(target, source) {
+  const propArr = [];
+
+  for (const property in source) {
+    propArr.push(property);
+  }
+
+  for (const propElement of propArr) {
+    if (!Object.hasOwn(target, propElement)) {
+      target[propElement] = source[propElement];
+    }
+  }
+}
```

Comment:
We can accomplish this using one `for` loop!
```
function defaults(target, source) {
  for (const key in source) {
    if (typeof target[key] === 'undefined') {
      target[key] = source[key];
    }
  }
}
```

Reply:
```
function defaults(target, source) {
  for (const key in source) {
    if (target[key] === undefined) {
      target[key] = source[key];
    }
  }
}
```

Is the typeof operator required in the if statement?

Reply:
Yep! Without `typeof`, it would just be `undefined` but `typeof` returns a string of what data type it is, so it will return `'undefined'`. So without `typeof` you would be trying to check if `undefined === 'undefined'` which would be false

Reply:
Got it. Thank you!